### ISSUE TITLE
Cannot {x:Bind} to property of named XAML control in C++/WinRT

### ISSUE THREAD

**Issue by sylveon** [microsoft/microsoft-ui-xaml#2721]
**Describe the bug**
The autogenerated binding code tries to find the named XAML control through the projected type rather than the implementation type, resulting in a compile-time error.

**Steps to reproduce the bug**
1. Create a new C++/WinRT XAML project
2. Add `<winrt/Windows.UI.Xaml.Media.h>` to the precompiled headers (to allow SolidColorBrush to be usable)
3. Add the following XAML code somewhere in the MainPage
```xaml
<StackPanel>
    <ColorPicker x:Name="Picker" IsAlphaEnabled="True" IsMoreButtonVisible="True" IsColorPreviewVisible="False" />

    <Rectangle Width="100" Height="100">
        <Rectangle.Fill>
            <SolidColorBrush Color="{x:Bind Picker.Color}" />
        </Rectangle.Fill>
    </Rectangle>
</StackPanel>
```
4. Build the code

**Expected behavior**
The code builds fine.

**Version Info**

NuGet package version: [Microsoft.Windows.CppWinRT 2.0.200615.7]

| Windows 10 version                  | Saw the problem? |
| :--------------------------------- | :-------------------- |
| Insider Build (xxxxx)              |    |
| May 2020 Update (19041)            | Yes   |
| November 2019 Update (18363)       |    |
| May 2019 Update (18362)            |    |
| October 2018 Update (17763)        |    |
| April 2018 Update (17134)          |    |
| Fall Creators Update (16299)       |    |
| Creators Update (15063)            |    |

| Device form factor | Saw the problem? |
| :----------------- | :--------------- |
| Desktop            | Yes |
| Xbox               |  |
| Surface Hub        |  |
| IoT                |  |

**Additional context**
The fix should be quite simple. Here's the wrong autogenerated code:
```cpp
void Update_(::winrt::BlankApp1::MainPage obj, int32_t phase)
{
    if (obj)
    {
        if ((phase & (NOT_PHASED | (1 << 0))) != 0)
        {
            Update_Picker(obj.Picker(), phase);
        }
    }
}
```

A correct version would be:
```cpp
void Update_(::winrt::BlankApp1::MainPage obj, int32_t phase)
{
    if (obj)
    {
        if ((phase & (NOT_PHASED | (1 << 0))) != 0)
        {
            Update_Picker(obj.as<::winrt::BlankApp1::implementation::MainPage>().Picker(), phase);
        }
    }
}
```

Alternatively, I believe it would be possible to instead have `Update_` directly take the implementation type:
```cpp
void Update_(::winrt::BlankApp1::implementation::MainPage obj, int32_t phase)
{
    if (obj)
    {
        if ((phase & (NOT_PHASED | (1 << 0))) != 0)
        {
            Update_Picker(obj.Picker(), phase);
        }
    }
}
```
by adjusting the generated struct declaration to refer to the implementation type, like this:
```cpp
    template <typename D, typename ... I>
    struct MainPageT<D, I...>::MainPage_obj1_Bindings
        : public ::winrt::BlankApp1::implementation::ReferenceTypeXamlBindings<::winrt::BlankApp1::implementation::MainPage, ::winrt::BlankApp1::implementation::XamlBindingTrackingBase>
{
```

**Comment by sylveon** [comment]
Additionally, looks like neither Binding works, the rectangle is just permanently black 😞

**Comment by JeanRoca** [comment]
Hey @sylveon would adding `ColorPicker Picker;​​` to your IDL file fix this issue?

**Comment by sylveon** [comment]
Yes, it does, but I don't want to expose it to my consumers like that.

**Comment by JeanRoca** [comment]
Oh I see that makes sense thank you for clarifying. In that case, there are ways to filter metadata, so that you don't expose this to customers. Adding @kennykerr and @benjkuhn who may be able to help workaround this while we wait on a real fix.

**Comment by evelynwu-msft** [comment]
Fix will be available in WinAppSDK 1.4.

**Comment by sylveon** [comment]
That's great news! Any chance at backporting to UWP?

**Comment by evelynwu-msft** [comment]
It's very unlikely given that a workaround (as ugly as it is) exists.

**Comment by duncanmacmichael** [comment]
Fixed in 1.4.0. @bpulliam please close this issue as I cannot. :)

**Comment by sylveon** [comment]
Not fixed in system XAML :(
